libs/community/tests/unit_tests/document_loaders/test_directory.py

```diff
@@ -56,3 +57,35 @@ def test_exclude_ignores_matching_files(tmp_path: Path) -> None:
 def test_exclude_as_string_converts_to_sequence() -> None:
     loader = DirectoryLoader("./some_directory", exclude="*.py")
     assert loader.exclude == ("*.py",)
+
+
+def test_directory_loader_glob_multiple() -> None:
+    """Verify that globbing multiple patterns in a list works correctly."""
+
+    path_to_examples = "tests/examples/"
+    list_extensions = [".rst", ".txt"]
+    list_globs = [f"**/*{ext}" for ext in list_extensions]
+    is_file_type_loaded = {ext: False for ext in list_extensions} 
+
+    loader = DirectoryLoader(
+        path=path_to_examples,
+        glob=list_globs,
+        loader_cls=TextLoader
```

Comment:
Can we use CustomLoader here so as to not change the imports? @baskaryan

Reply:
I will take a look at replacing TextLoader with CustomLoader briefly. Should be able to make the suggested change on my fork shortly. Thank you